### PR TITLE
compat: image create: handle platform correctly

### PR DESCRIPTION
**Pull request by vrothberg**
Handle the platform parameter correctly.  The parameter was only parsed
in presence of credentials and the code was a bit complex.  Also add a
regression test.

Fixes: #10977
Signed-off-by: Valentin Rothberg <rothberg@redhat.com>

### CODE REVIEW

File: test/apiv2/10-images.at [hunk 1]
```diff
@@ -45,6 +45,10 @@ t POST "images/create?fromImage=alpine" 200 .error~null .status~".*Download comp
 
 t POST "images/create?fromImage=alpine&tag=latest" 200
 
+# 10977 - handle platform parameter correctly
+t POST "images/create?fromImage=alpine&platform=linux/arm64" 200
+is "$($PODMAN_BIN inspect --format '{{.Architecture}}' alpine)" "arm64" "platform parameter works"
```
Comment (vrothberg) [inline test/apiv2/10-images.at:50]: Note: `$PODMAN_BIN` was necessary to actually get the output of the command.

**Comment by rhatdan** [comment]
LGTM

File: pkg/api/handlers/compat/images.go [hunk 2]
```diff
@@ -266,41 +266,26 @@ func CreateImageFromImage(w http.ResponseWriter, r *http.Request) {
...
-				pullOptions.Variant = platformSpecs[2]
-			}
-		}
 	}
 	pullOptions.Writer = os.Stderr // allows for debugging on the server
 
-	progress := make(chan types.ProgressProperties)
+	// Handle the platform.
+	platformSpecs := strings.Split(query.Platform, "/")
+	if len(platformSpecs) > 1 {
```
Comment (cdoern) [inline pkg/api/handlers/compat/images.go:280]: I think you can pass an OS (technically) without a platform even though it doesn't make much sense that is why I checked explicitly for length =1, 2 or 3

File: test/apiv2/10-images.at [hunk 3]
```diff
@@ -45,6 +45,11 @@ t POST "images/create?fromImage=alpine" 200 .error~null .status~".*Download comp
 
 t POST "images/create?fromImage=alpine&tag=latest" 200
 
+# 10977 - handle platform parameter correctly
+t POST "images/create?fromImage=alpine&platform=linux/arm64" 200
+t GET  "images/alpine/json" 200
+is "$(echo $output | jq .Architecture)" "\"arm64\"" "platform parameter works"
```
Comment (Luap99) [inline test/apiv2/10-images.at:51]: Can you please use the following like the other tests:
```suggestion
t GET  "images/alpine/json" 200 \
  .Architecture=arm64
```
This looks much better in the test log.

File: pkg/api/handlers/compat/images.go (hunk 2 above)
Comment (vrothberg) [reply pkg/api/handlers/compat/images.go:280]: Good catch, I'll fix that.

**Pull request merged**